applications/application-registry.md

```diff
@@ -0,0 +1,205 @@
+- **FTE:**  1.2
+- **Costs:** 106,081 USD *
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | MIT |
+| 0b. | Documentation | We will provide documentation as part of the source code and a handbook on how wallet or application developers can use the Application Registry smart contract. It will contain how to build and deploy the code, how to interact with the contract, how to use EOSJS. |
+| 0c. | Testing Guide | We will provide automated Smart Contract test cases using DUNE. |
+| 1. | Application Registry (Smart Contract) | The Smart Contract will be built in such a way that it considers the governance as defined in an earlier step and will allow the Application metadata to be stored on-chain. The Smart Contract will be designed to be deployed on any EOSIO chain. |  
+
+\* Additional sub-milestones may be required to support effective cash flow for the project team
+
+
+### Milestone 3 — Build Application Registry
+
+- **Estimated Duration:** 7 weeks
+- **FTE:**  0.5
+- **Costs:** 23,783 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | MIT |
+| 0b. | Documentation | We will provide documentation on the available indexes and what data is being represented. |
+| 0c. | Testing Guide | We will provide test scripts that can verify that the indexes are populated and return results. |
+| 1. | Application Registry (Smart Contract) | The Smart Contract will be built in such a way that it considers the governance as defined in an earlier step and will allow the Application metadata to be stored on-chain. The Smart Contract will be designed to be deployed on any EOSIO chain. |  
+
+
+### Milestone 4 — Application Registry Web Frontend
+
+- **Estimated Duration:** 9 weeks
+- **FTE:**  1.4
+- **Costs:** 81,115 USD *
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | MIT |
+| 0b. | Documentation | We will provide both inline documentation of the code and a basic tutorial that explains how the various functions of the Frontend work for each user type. These user types are app developers, validators, end users and administrators. |
+| 0c. | Testing Guide | We will provide test procedures on how to verify that all functionality is working as expected. |
+| 1. | Application Registry Web Frontend | This is the Web Portal where users can browse the Application Registry. Application developers will be able to register their applications and will be able to make changes to existing Application Registry entries. Validators and Administrators will be able to validate Application Registry requests. Users are also able to “report” applications that are not behaving as they should, in order for a Validator to take the necessary action, which could be to warn the Application developer about this unwanted behavior or to remove the Application from the Registry completely. |  
```

Comment:
Are "reports" logged on-chain or off? How will these reports be handled in terms of transparency, anonymity (whistleblowing), etc?

Reply:
We recommend off-chain for privacy reasons and some kind of case management workflow. If needed we can set this up in a project management tool, and also include a place for logging future improvement and maintenance requests. We are open to other suggestions and this can be tabled during the discovery sessions.